### PR TITLE
Handle errors when updating changelog

### LINKED ISSUE #378: Instant crash on first run if no Internet access
I'm using the F-Droid build of Jerboa 0.0.30 on GrapheneOS (Android 13) on a Pixel 7.

**Obviously, I realise that Jerboa by its very nature requires Internet access to work.** However, I tend to try launching new apps without Internet access to see how they react. In Jerboa's case, the answer is that it crashes instantly on launch while displaying the "Support Jerboa" box, without any explanation, when trying to connect to GitHub. (Presumably while [trying to access the RELEASES.md file](https://github.com/dessalines/jerboa/blob/0.0.30/app/src/main/java/com/jerboa/ui/components/common/Dialogs.kt#L303-L311).)

In this case, I would expect Jerboa to instead just not be able to load the changelog at all and to let the user know, but let the user pass if they were okay with doing so. Crashing instantly means that it's not very obvious at all what the problem is.

The exception is as follows (in this case, with Wifi and Data access turned off; a different error occurs if the Internet access is blocked completely by using AFWall+, but the resolution is the same - allow Internet access):

```
type: crash
osVersion: google/panther/panther:13/TQ2A.230505.002/2023050100:user/release-keys
package: com.jerboa:30
process: com.jerboa
processUptime: 259 + 235 ms

java.net.UnknownHostException: Unable to resolve host "raw.githubusercontent.com": No address associated with hostname
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:147)
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
	at java.net.InetAddress.getAllByName(InetAddress.java:1152)
	at m1.f.i(Unknown Source:5)
	at x7.a.a(Unknown Source:451)
	at y7.g.b(Unknown Source:138)
	at v7.a.a(Unknown Source:120)
	at y7.g.b(Unknown Source:138)
	at y7.a.a(Unknown Source:174)
	at y7.g.b(Unknown Source:138)
	at y7.h.a(Unknown Source:153)
	at y7.g.b(Unknown Source:138)
	at x7.k.i(Unknown Source:95)
	at x7.k.f(Unknown Source:41)
	at i5.u.h(Unknown Source:98)
	at y6.a.j(Unknown Source:8)
	at p7.f0.run(Unknown Source:104)
	at kotlinx.coroutines.internal.g.run(Unknown Source:11)
	at kotlinx.coroutines.scheduling.i.run(Unknown Source:2)
	at kotlinx.coroutines.scheduling.a.run(Unknown Source:76)
	Suppressed: e4.f: [androidx.compose.ui.platform.s1@9d9b799, g0.f@f9fa5e, p1{Cancelling}@76c6e3f, Dispatchers.IO]
Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
	at libcore.io.Linux.android_getaddrinfo(Native Method)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:133)
	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:222)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:133)
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:135)
	... 19 more

```

### PR DESCRIPTION
**Pull request by marcellogalhardo**
* Handle errors when updating changelog;
* Move fetching logic from view to `AppSettingsRepository`;
* Replace `scope.launch` by `LaunchedEffect` to avoid unnecessary calls during recompositions;
* Use `getValue` delegate on states to reduce verbosity (no need for `state.value = ` each time);

Fixes #378

### CODE REVIEW

File: app/src/main/java/com/jerboa/db/AppDB.kt [hunk 1]
```diff
@@ -3,15 +3,35 @@ package com.jerboa.db
 import android.content.ContentValues
 import android.content.Context
 import android.database.sqlite.SQLiteDatabase.CONFLICT_IGNORE
+import android.util.Log
 import androidx.annotation.WorkerThread
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
-import androidx.room.*
```
Comment (marcellogalhardo) [inline app/src/main/java/com/jerboa/db/AppDB.kt:11]: Sorry for that, my IDE is configured to not use wildcards. Will undo it.

File: app/src/main/java/com/jerboa/db/AppDB.kt [hunk 2]
```diff
@@ -173,7 +182,14 @@ class AccountRepository(private val accountDao: AccountDao) {
...
 // Declares the DAO as a private property in the constructor. Pass in the DAO
 // instead of the whole database, because you only need access to the DAO
-class AppSettingsRepository(private val appSettingsDao: AppSettingsDao) {
+class AppSettingsRepository(
+    private val appSettingsDao: AppSettingsDao,
+    private val httpClient: OkHttpClient = OkHttpClient(),
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) {
+
+    private val _changelog = MutableStateFlow("")
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/db/AppDB.kt:191]: No reason to use a stateflow here, just use mutablestate. Its also the convention I've gone with in the `autogen_types` branch, in all the viewModels.

Comment (marcellogalhardo) [reply app/src/main/java/com/jerboa/db/AppDB.kt:196]: Do you use `State<T>` inside a `Repository`? That is new to me. I will update the branch later.

**Review by dessalines — APPROVED** [review]

**Comment by twizmwazin** [comment]
@marcellogalhardo Can you enable the option to allow maintainers to edit your pull request? We're ready to merge but we don't merge if the PR is out of date with main. If we can edit the branch I can get this merged tonight.

**Comment by marcellogalhardo** [comment]
> @marcellogalhardo Can you enable the option to allow maintainers to edit your pull request? We're ready to merge but we don't merge if the PR is out of date with main. If we can edit the branch I can get this merged tonight.

@twizmwazin you should have the rights now! Sorry for the inconvenience.

**Comment by dessalines** [comment]
Mmmk I got this one updated, it should get merged soon.

**Pull request merged**